Overhaul Voyage tab

rglara opened the pull request:
**TLDR**
* Add commas to counters in main header (chron, dil, etc.)
* Facelift for Voyage tab
  * Larger icons
  * Tables instead of lists
  * "Pretty" groups
  * Add dark background to skill icons on light theme

I think I covered all the states of voyages (not started, ongoing, done, etc.), but I'm waiting on my own voyages to see how they react to ensure it looks good for all phases. 

**Longer conversation**
I couldn't see the skill icons when using the light theme (light blue on white). But the dark theme looked decent. So, I wanted to add a dark background to them on light theme. I just took the first screen that had a few of them (aka Voyage tab).

I took a few liberties with the design, so any feedback is appreciated. Especially if you had a vision in mind.

Side conversations (these are widespread, but could be knocked out in other PRs):
* Tabs or 4 spaces for indenting?
  * It isn't consistent
* CRLF vs. LF
  * Looks like Windows and non-Windows folks pitch in here, but do we want to standardize on LF and do some `.gitattributes` to keep it consistent?

rglara committed:
comma separators in numbers in header

rglara committed:
Update Voyage Log layouts

Add DarkThemeContext and Provider

rglara committed:
Update Voyage complete log layout

Minor updates to remove console errors

rglara committed:
update layouts on Voyage Crew Select

split out Voyage Skills Readout into its own component

rglara committed:
convert Voyage Log to use new shared component

paulbilnoski commented:
Thanks for the PR! Will review more closely soon, but I did want to comment on the whitespace issues.
I don't have a big preference between tabs or spaces, but still lean towards spaces (hence inconsistencies everywhere). Originally most files used tabs with some inconsistency and the files I started have spaces. Where a file is inconsistent and gets edited for a commit is an opportunity to clean it up to spaces-only, but I want to avoid a pass at all files just to fix whitespace.
As for line endings, I prefer the simpler unix endings since they behave better in most tools though I typically develop on Windows and have user settings to govern it.
I also prefer to have a newline at the end of the file, again to aid in various tool use.
Configuration to keep the repo consistent would be fine to add.

paulbilnoski commented on src/components/voyage/VoyageLogEntry.tsx:
```diff
@@ -1,51 +1,54 @@
          if (entry.crew) {
-            let rc = STTApi.roster.find(rosterCrew => rosterCrew.symbol == entry.crew[0]);
-            if (rc) entry.crewIconUrl = rc.iconUrl;
+            entry.crewIconUrls = [];
+            for(let i = 0; i < entry.crew.length; i += 1) {
+               let rc = STTApi.roster.find(rosterCrew => rosterCrew.symbol == entry.crew[i]);
+               entry.crewIconUrls.push(rc ? rc.iconUrl ?? '' : '');
+            }
```
This could be simpler as a one-liner via map:
entry.crewIconUrls = entry.crew.map(ec => STTApi.roster.find(rosterCrew => rosterCrew.symbol == ec)).map(rc => rc ? rc.iconUrl ?? '' : '');

paulbilnoski commented on src/components/voyage/VoyageLog.tsx:
```diff
@@ -126,17 +127,16 @@ export const VoyageLog = (props:{}) => {
             resizable: false,
             accessor: (row: any) => row.full_name,
-            Cell: (p: any) => <img className={spriteClass} src={p.original.iconUrl} height={25} />
+            Cell: (p: any) => <img className={`image-fit ${spriteClass}`} src={p.original.iconUrl} />
```
The icons in the reward table are too large. 25-30px would be better than the original image size, which sometimes varies.

paulbilnoski commented on src/components/voyage/VoyageLogEntry.tsx:
```diff
@@ -1,51 +1,54 @@
+         }
+         let textClass = 'vle-text';
+         if (entry.skill_check && (index + 1) === this.props.log.length) {
+            textClass = `${textClass} ui label ${entry.skill_check.passed ? 'green' : 'red'}`
```
Using "ui label green" or "ui label red" makes this look like a button, but does not function as one. Please adjust the styling here.

paulbilnoski commented on src/components/voyage/VoyageCrewSelect.tsx:
```diff
@@ -620,43 +628,41 @@ const BestCrew = (props : {
-				</ul>
-
+				{
+					props.crewSelection && props.crewSelection.length &&
```
Prints an errant "0" when no crew is selected. Also crewSelection will not be undefined, so this can instead be changed to
{props.crewSelection.length > 0 && <VoyageSkillsReadout ... />}

paulbilnoski reviewed (changes requested):
Wow, it is great to have someone that knows CSS help out, very nice UI updates indeed.
There are a few minor issues to address I added in other comments (some before I figured out a "review" in github).
There are also a few places (I marked one in a comment) where you can use JS forEach or map on an array instead of a loop counter variable. That is more an FYI than "requires fixing", but might get transformed in the future as I touch files and see them.

paulbilnoski commented:
To test, there is a flag in STTApi "_useMock" that you can enable to use stored data instead of pulling your account data live. Just dump the contents returned by the servers to file in a mock/ dir under the app path and the app will use that instead. You can store data from voyages at various points to test later or edit the file as you like to coerce the app into particular behavior.

Also, if you would like, other areas that I think badly need UI updates are:
- The "stats" section at the top of the voyage log
- The EventLeaderboard display of the EventHelperPage (currently uses more unordered-lists)
- Cryo collections page

rglara commented:
> As for line endings, I prefer the simpler unix endings since they behave better in most tools though I typically develop on Windows and have user settings to govern it.
> I also prefer to have a newline at the end of the file, again to aid in various tool use.
> Configuration to keep the repo consistent would be fine to add.

I'll see about another PR for converting the CRLF to LF (to target no content changes to make it easier to review)

rglara commented:
Updated for comments made; please re-review

rglara committed:
fix for loop to .map

fix ui label to header (not like a button anymore)

remove rendering of errant zero

tabs to spaces in VoyageCrewSelect.tsx

reduce size of icons in rewards table

paulbilnoski commented:
Looks nice, thanks very much for the updates and the PR!
